packages/app/src/visualizer/utils.ts

```diff
@@ -88,16 +92,17 @@ function getSupportedCoreVis(
 
 function getSupportedNxVis(
   entity: Entity,
-  attrValueStore: AttrValuesStore
+  attrValuesStore: AttrValuesStore
 ): VisDef[] {
-  if (!isGroup(entity) || !isNxDataGroup(entity, attrValueStore)) {
+  if (!isGroup(entity) || !isNxDataGroup(entity, attrValuesStore)) {
     return [];
   }
 
   assertGroupWithChildren(entity);
-  const dataset = findSignalDataset(entity, attrValueStore);
+  const dataset = findSignalDataset(entity, attrValuesStore);
+  const axisDatasets = findAssociatedDatasets(entity, 'axes', attrValuesStore);
```

Comment:
The particularity of NxScatter is that the two axes datasets must be defined. Therefore, I need to check for them here.

Reply:
This doesn't trigger any requests, right? Because the attribute and the metadata of the children should already be fetched at this point.

That being said, it might still be worth moving this line inside the new `if` below, and splitting the `if` in two: a first check on the dimension of `dataset` and then on the axis datasets:

```ts
  if (hasNumDims(dataset, 1)) {
    const axisDatasets = findAssociatedDatasets(entity, 'axes', attrValuesStore);

    if (axisDatasets.length === 2 && axisDatasets.every(d => d && hasNumDims(d, 1)) {
      return [NEXUS_VIS[NexusVis.NxScatter]];
    }
  }
```

I would maybe also move this check below the interpretation checks so `findAssociatedDatasets` is not called for simple `spectrum` NXdata groups.